### ISSUE TITLE
t5.pl failing on master with encryption enabled

### ISSUE THREAD

**Issue by DeepDiver1975** [owncloud/core#15957]
https://ci.owncloud.org/job/client-test-matrix-linux-8.1-enc/18/compiler=g++,label=master,mirallBranch=master,serverVersion=owncloud8.1,tc=t5/console

More analysis to be done ....

**Comment by DeepDiver1975** [comment]
````
{"reqId":"FTT5fFLxU0VaRNWT7ldF","remoteAddr":"::1","app":"PHP","message":"fread(): Length parameter must be greater than 0 at \/home\/deepdiver\/Development\/ownCloud\/core\/3rdparty\/icewind\/streams\/src\/Wrapper.php#67","level":3,"time":"2015-04-30T12:01:19+00:00","method":"DELETE","url":"\/oc\/core\/remote.php\/webdav\/t1-20150430140039-78\/share_source\/newDir\/a_file.bin"}
{"reqId":"FTT5fFLxU0VaRNWT7ldF","remoteAddr":"::1","app":"webdav","message":"Exception: {\"Message\":\"Encryption library: Decryption (symmetric) of content failed: error:0607A082:digital envelope routines:EVP_CIPHER_CTX_set_key_length:invalid key length\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/encryption\\\/lib\\\/crypto\\\/crypt.php(294): OCA\\\\Encryption\\\\Crypto\\\\Crypt->decrypt(false, false, 'pGvHSxode16FPIh...', 'AES-128-CFB')\\n#1 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/encryption\\\/lib\\\/crypto\\\/encryption.php(259): OCA\\\\Encryption\\\\Crypto\\\\Crypt->symmetricDecryptFileContent('---------------...', 'pGvHSxode16FPIh...', 'AES-128-CFB')\\n#2 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/stream\\\/encryption.php(432): OCA\\\\Encryption\\\\Crypto\\\\Encryption->decrypt('---------------...')\\n#3 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/stream\\\/encryption.php(278): OC\\\\Files\\\\Stream\\\\Encryption->readCache()\\n#4 [internal function]: OC\\\\Files\\\\Stream\\\\Encryption->stream_read(8192)\\n#5 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/helper.php(611): fread(Resource id #54, 8192)\\n#6 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/common.php(549): OC_Helper::streamCopy(Resource id #54, Resource id #55)\\n#7 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/local.php(371): OC\\\\Files\\\\Storage\\\\Common->copyFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#8 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/common.php(573): OC\\\\Files\\\\Storage\\\\Local->copyFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...', true)\\n#9 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/local.php(389): OC\\\\Files\\\\Storage\\\\Common->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#10 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(526): OC\\\\Files\\\\Storage\\\\Local->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#11 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(526): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#12 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/view.php(650): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#13 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/files_trashbin\\\/lib\\\/trashbin.php(192): OC\\\\Files\\\\View->rename('\\\/files\\\/t1-20150...', '\\\/files_trashbin...')\\n#14 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/files_trashbin\\\/lib\\\/storage.php(100): OCA\\\\Files_Trashbin\\\\Trashbin::move2trash('t1-201504301400...')\\n#15 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/encryption.php(212): OCA\\\\Files_Trashbin\\\\Storage->unlink('newDir\\\/a_file.b...')\\n#16 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/view.php(938): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption->unlink('newDir\\\/a_file.b...')\\n#17 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/view.php(573): OC\\\\Files\\\\View->basicOperation('unlink', '\\\/t1-20150430140...', Array)\\n#18 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(246): OC\\\\Files\\\\View->unlink('\\\/t1-20150430140...')\\n#19 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Tree.php(179): OC\\\\Connector\\\\Sabre\\\\File->delete()\\n#20 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(287): Sabre\\\\DAV\\\\Tree->delete('t1-201504301400...')\\n#21 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpDelete(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#22 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#23 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:DELETE', Array)\\n#24 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#25 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/files\\\/appinfo\\\/remote.php(81): Sabre\\\\DAV\\\\Server->exec()\\n#26 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/remote.php(84): require_once('\\\/home\\\/deepdiver...')\\n#27 {main}\",\"File\":\"\\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/encryption\\\/lib\\\/crypto\\\/crypt.php\",\"Line\":351}","level":4,"time":"2015-04-30T12:01:19+00:00","method":"DELETE","url":"\/oc\/core\/remote.php\/webdav\/t1-20150430140039-78\/share_source\/newDir\/a_file.bin"}

**Comment by DeepDiver1975** [comment]
```json


{
  "reqId": "FTT5fFLxU0VaRNWT7ldF",
  "remoteAddr": "::1",
  "app": "PHP",
  "message": "fread(): Length parameter must be greater than 0 at \/home\/deepdiver\/Development\/ownCloud\/core\/3rdparty\/icewind\/streams\/src\/Wrapper.php#67",
  "level": 3,
  "time": "2015-04-30T12:01:19+00:00",
  "method": "DELETE",
  "url": "\/oc\/core\/remote.php\/webdav\/t1-20150430140039-78\/share_source\/newDir\/a_file.bin"
}

**Comment by DeepDiver1975** [comment]
```json


{
  "reqId": "FTT5fFLxU0VaRNWT7ldF",
  "remoteAddr": "::1",
  "app": "webdav",
  "message": "Exception: {\"Message\":\"Encryption library: Decryption (symmetric) of content failed: error:0607A082:digital envelope routines:EVP_CIPHER_CTX_set_key_length:invalid key length\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/encryption\\\/lib\\\/crypto\\\/crypt.php(294): OCA\\\\Encryption\\\\Crypto\\\\Crypt->decrypt(false, false, 'pGvHSxode16FPIh...', 'AES-128-CFB')\\n#1 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/encryption\\\/lib\\\/crypto\\\/encryption.php(259): OCA\\\\Encryption\\\\Crypto\\\\Crypt->symmetricDecryptFileContent('---------------...', 'pGvHSxode16FPIh...', 'AES-128-CFB')\\n#2 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/stream\\\/encryption.php(432): OCA\\\\Encryption\\\\Crypto\\\\Encryption->decrypt('---------------...')\\n#3 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/stream\\\/encryption.php(278): OC\\\\Files\\\\Stream\\\\Encryption->readCache()\\n#4 [internal function]: OC\\\\Files\\\\Stream\\\\Encryption->stream_read(8192)\\n#5 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/helper.php(611): fread(Resource id #54, 8192)\\n#6 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/common.php(549): OC_Helper::streamCopy(Resource id #54, Resource id #55)\\n#7 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/local.php(371): OC\\\\Files\\\\Storage\\\\Common->copyFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#8 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/common.php(573): OC\\\\Files\\\\Storage\\\\Local->copyFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...', true)\\n#9 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/local.php(389): OC\\\\Files\\\\Storage\\\\Common->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#10 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(526): OC\\\\Files\\\\Storage\\\\Local->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#11 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(526): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#12 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/view.php(650): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->moveFromStorage(Object(OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption), 'newDir\\\/a_file.b...', 'files_trashbin\\\/...')\\n#13 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/files_trashbin\\\/lib\\\/trashbin.php(192): OC\\\\Files\\\\View->rename('\\\/files\\\/t1-20150...', '\\\/files_trashbin...')\\n#14 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/files_trashbin\\\/lib\\\/storage.php(100): OCA\\\\Files_Trashbin\\\\Trashbin::move2trash('t1-201504301400...')\\n#15 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/encryption.php(212): OCA\\\\Files_Trashbin\\\\Storage->unlink('newDir\\\/a_file.b...')\\n#16 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/view.php(938): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Encryption->unlink('newDir\\\/a_file.b...')\\n#17 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/files\\\/view.php(573): OC\\\\Files\\\\View->basicOperation('unlink', '\\\/t1-20150430140...', Array)\\n#18 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/lib\\\/private\\\/connector\\\/sabre\\\/file.php(246): OC\\\\Files\\\\View->unlink('\\\/t1-20150430140...')\\n#19 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Tree.php(179): OC\\\\Connector\\\\Sabre\\\\File->delete()\\n#20 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(287): Sabre\\\\DAV\\\\Tree->delete('t1-201504301400...')\\n#21 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpDelete(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#22 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#23 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(469): Sabre\\\\Event\\\\EventEmitter->emit('method:DELETE', Array)\\n#24 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#25 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/files\\\/appinfo\\\/remote.php(81): Sabre\\\\DAV\\\\Server->exec()\\n#26 \\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/remote.php(84): require_once('\\\/home\\\/deepdiver...')\\n#27 {main}\",\"File\":\"\\\/home\\\/deepdiver\\\/Development\\\/ownCloud\\\/core\\\/apps\\\/encryption\\\/lib\\\/crypto\\\/crypt.php\",\"Line\":351}",
  "level": 4,
  "time": "2015-04-30T12:01:19+00:00",
  "method": "DELETE",
  "url": "\/oc\/core\/remote.php\/webdav\/t1-20150430140039-78\/share_source\/newDir\/a_file.bin"
}

**Comment by DeepDiver1975** [comment]
t5.pl is failing because of these two exceptions above

**Comment by DeepDiver1975** [comment]
this was fixed with #15968
